smoke/LeafDevice/details/Details.cs

```diff
@@ -50,12 +50,16 @@ protected Task InstallCaCertificate()
         protected async Task ConnectToEdgeAndSendData()
         {
             Microsoft.Azure.Devices.IotHubConnectionStringBuilder builder = Microsoft.Azure.Devices.IotHubConnectionStringBuilder.Create(this.iothubConnectionString);
-            string leafDeviceConnectionString = $"HostName={builder.HostName};DeviceId={this.deviceId};SharedAccessKey={this.context.Device.Authentication.SymmetricKey.PrimaryKey};gatewayHostName={this.edgeHostName}";
+            string leafDeviceConnectionString = $"HostName={builder.HostName};DeviceId={this.deviceId};SharedAccessKey={this.context.Device.Authentication.SymmetricKey.PrimaryKey};GatewayHostName={this.edgeHostName}";
+            
+            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(leafDeviceConnectionString, Microsoft.Azure.Devices.Client.TransportType.Mqtt);
+            Console.WriteLine("Leaf Device client created.");
 
-            DeviceClient deviceClient = DeviceClient.CreateFromConnectionString(leafDeviceConnectionString, Microsoft.Azure.Devices.Client.TransportType.Amqp);
 
             var message = new Microsoft.Azure.Devices.Client.Message(Encoding.ASCII.GetBytes($"Message from Leaf Device. MsgGUID: {this.context.MessageGuid}"));
+            Console.WriteLine($"Trying to send the message to '{this.edgeHostName}'");
             await deviceClient.SendEventAsync(message);
+            Console.WriteLine($"Message Sent. ");
```

Comment:
When it comes to printing messages to the console, my approach with QuickStart was to print only the minimal amount of information needed so that the user knows the app is using the arguments that were passed in.

In this case, the app will always try to send a message, and the message doesn't change based on the arguments that are passed in. If the app prints "Success!" before it exits, then we know the message was sent. So you don't need these extra Console.Writeline calls.

Reply:
There extra Console Lines helped a lot to debug. I think that, in case of failure, having minimal amount of message doesn't help that much. 
Since this is a tool used for E2E (or deployment), having these additional information helps debugging.